apps/Typescript-ObjC/App.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ *
+ * @format
+ */
+
+import React, {useEffect} from 'react';
+import type {PropsWithChildren} from 'react';
+import {
+  SafeAreaView,
+  ScrollView,
+  StatusBar,
+  StyleSheet,
+  Text,
+  Button,
+  useColorScheme,
+  View,
+} from 'react-native';
+
+import {
+  Colors,
+  DebugInstructions,
+  Header,
+  LearnMoreLinks,
+  ReloadInstructions,
+} from 'react-native/Libraries/NewAppScreen';
+
+import {
+  CustomerIO,
+  CustomerIOEnv,
+  CustomerioConfig,
+  CioLogLevel,
+} from 'customerio-reactnative';
+
+type SectionProps = PropsWithChildren<{
+  title: string;
+}>;
+
+function Section({children, title}: SectionProps): JSX.Element {
+  const isDarkMode = useColorScheme() === 'dark';
+  return (
+    <View style={styles.sectionContainer}>
+      <Text
+        style={[
+          styles.sectionTitle,
+          {
+            color: isDarkMode ? Colors.white : Colors.black,
+          },
+        ]}>
+        {title}
+      </Text>
+      <Text
+        style={[
+          styles.sectionDescription,
+          {
+            color: isDarkMode ? Colors.light : Colors.dark,
+          },
+        ]}>
+        {children}
+      </Text>
+    </View>
+  );
+}
+
+function App(): JSX.Element {
+  useEffect(() => {
+    const env = new CustomerIOEnv();
+    env.siteId = 'YourSiteId';
+    env.apiKey = 'YourAPIKey';
+
+    const config = new CustomerioConfig();
+    config.logLevel = CioLogLevel.debug;
+
+    CustomerIO.initialize(env, config);
+  }, []);
+
+  const isDarkMode = useColorScheme() === 'dark';
+
+  const backgroundStyle = {
+    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
+  };
+
+  return (
+    <SafeAreaView style={backgroundStyle}>
+      <StatusBar
+        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
+        backgroundColor={backgroundStyle.backgroundColor}
+      />
+      <ScrollView
+        contentInsetAdjustmentBehavior="automatic"
+        style={backgroundStyle}>
+        <Header />
+        <View
+          style={{
+            backgroundColor: isDarkMode ? Colors.black : Colors.white,
+          }}>
+          <Button
+            title="Press me"
+            onPress={() => {
+              CustomerIO.identify('typescript@customer.io', {});
+              CustomerIO.track('track event', {});
+              CustomerIO.screen('screen tracked', {});
+            }}
```

Comment:
```suggestion
       const handlePress = (): void => {
    CustomerIO.identify('typescript@customer.io', {});
    CustomerIO.track('track event', {});
    CustomerIO.screen('screen tracked', {});
  };
```

Maybe extract it? otherwise, this piece of code will keep on getting bigger.